core/src/main/java/org/infinispan/commands/CommandsFactory.java

```diff
@@ -460,7 +460,7 @@
 
    <K, V, R> ReadOnlyKeyCommand<K, V, R> buildReadOnlyKeyCommand(K key, Function<ReadEntryView<K, V>, R> f);
 
-   <K, V, R> ReadOnlyManyCommand<K, V, R> buildReadOnlyManyCommand(Set<? extends K> keys, Function<ReadEntryView<K, V>, R> f);
+   <K, V, R> ReadOnlyManyCommand<K, V, R> buildReadOnlyManyCommand(Collection<? extends K> keys, Function<ReadEntryView<K, V>, R> f);
```

Comment:
Eh, I thought I put a comment somewhere, but it got lost...

When you build the collection as an user (and pass it to the functional API), it's useful to mark it as a set to prevent repetitions. However, during the internal processing you sometimes need to split the set according to keys' owners, and it's more convenient and effective to just put those subsets into `ArrayList`s.
Also the Set may use different equality/hash functions than `Object`'s default ones, so if you just create new `HashSet`s, you risk losing some keys as you add them there (if the `Object.equals()` matches for some of them.